django_cloud_tasks/models.py

```diff
@@ -0,0 +1,122 @@
+from typing import Optional
+from django.db import transaction
+from django.apps import apps
+
+from datetime import datetime
+from django.db import models
+from django_cloud_tasks import tasks
+from django_cloud_tasks.errors import PipelineError
+
+class Pipeline(models.Model):
+    name = models.CharField(max_length=50)
+
+    def start(self):
+        routines = self.routines.filter(next_routines__id__isnull=True)
+        if not routines.exists():
+            raise PipelineError("Not enought routines to start")
+
+        if routines.count() > 1:
+            raise PipelineError("Too many routines to start at once")
+
+        routines.first().enqueue()
+
+    def revert(self):
+        for routine in self.routines.all():
+            routine.revert()
+
+
+class Routine(models.Model):
+    class Statuses(models.TextChoices):
+        COMPLETED = ("completed", "Completed")
+        FAILED = ("failed", "Failed")
+        REVERTING = ("reverting", "Reverting")
+        REVERTED = ("reverted", "Reverted")
+        ABORTED = ("aborted", "Aborted")
```

Comment:
What's the first status? When the task has just been created but is not enqueued yet?

I think it's missing 3 more statuses here: `PENDING`, `SCHEDULED`, `RUNNING`

Reply:
Perfect, I going to add them. I thinking make `PENDING` the initial status